### PR TITLE
global: fix imports and remove search_engine dependency

### LINKED ISSUE #6: standardize travis.yml as the template of cookiecutter
We need to use a more "standard" travis.yml as in https://github.com/inveniosoftware/cookiecutter-invenio-module/blob/master/%7B%7B%20cookiecutter.project_shortname%20%7D%7D/.travis.yml

### LINKED ISSUE #11: RFC legacy: remove search_engine dependency
#### Problem
Currently, the module use legacy `search_engine` to:
- Api:
  - detach_tag_from_record
  - attach_tags_to_record
  - attach_tag_to_record
  - get_attached_tags_on_record
- Form:
  - validate_user_can_see_bibrec

#### Question
How can we remove the dependency?

@jirikuncar

### PR DESCRIPTION
**Pull request by hachreak**
* INCOMPATIBLE Removes dependency on legacy search_engine module.
  (addresses inveniosoftware/invenio#3233) (closes #11)

* Improves Travis configuration. (closes #6) (closes #7)

* Fixes user, record, collection dependencies.

### CODE REVIEW

File: invenio_tags/models.py [hunk 1]
```diff
@@ -23,14 +23,17 @@
...
 from invenio.ext.sqlalchemy import db
-from invenio.modules.accounts.models import User, Usergroup
-from invenio.modules.records.models import Record as Bibrec
 from invenio.utils.text import wash_for_xml
 
+from invenio_accounts.models import User
+
+from invenio_groups.models import Group
+
+from inveniod_records.models import Record
```
Comment (jirikuncar) [inline invenio_tags/models.py]: typo?

**Comment by jirikuncar** [comment]
@hachreak can you check and amend `setup.py` and `requirements-devel.txt`?

File: .travis.invenio.cfg [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+# EXTENSIONS = [
+# 'invenio.ext.sqlalchemy',
+# 'invenio.ext.restful',
+# 'invenio.ext.template',
+# 'flask_breadcrumbs:Breadcrumbs',
+# ]
+
+PACKAGES = [
+    'invenio.base',
```
Comment (jirikuncar) [inline .travis.invenio.cfg]: invenio_base + move at the end

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2013, 2014, 2015 CERN.
+# Copyright (C) 2015 CERN.
```
Comment (jirikuncar) [inline .travis.yml]: keep the copyright years

File: docs/_ext/flask_app.py [hunk 4]
```diff
@@ -24,8 +24,8 @@ def setup(sphinx):
     """Setup Sphinx object."""
     from flask import has_app_context
     from invenio.base.factory import create_app
-    PACKAGES = ['invenio.base', 'invenio.modules.accounts',
-                'invenio.modules.records', 'invenio_tags']
+    PACKAGES = ['invenio.base', 'invenio_accounts', 'invenio_groups',
```
Comment (jirikuncar) [inline docs/_ext/flask_app.py]: please fix the order and invenio_base

File: requirements-devel.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+-e git+git://github.com/mitsuhiko/werkzeug.git#egg=Werkzeug
```
Comment (jirikuncar) [inline requirements-devel.txt]: please include copyright header

File: setup.py [hunk 6]
```diff
@@ -96,8 +106,8 @@ def run_tests(self):
     install_requires=requirements,
     extras_require={
         'docs': [
-            'Sphinx',
-            'sphinx_rtd_theme'
+            'Sphinx>=1.3',
+            'sphinx_rtd_theme>=0.1.7'
         ],
     },
```
Comment (jirikuncar) [inline setup.py]: 'tests': test_requirements

**Comment by jirikuncar** [comment]
- [ ] babel.init
- [ ] extract messages (https://gist.github.com/jirikuncar/654560a5e4f5498ffa4f#file-gistfile1-txt-L27-L31)

File: requirements-devel.txt (hunk 5 above)
Comment (hachreak) [inline requirements-devel.txt]: done

**Comment by hachreak** [comment]
translation added.

**Comment by hachreak** [comment]
ping @jirikuncar @SamiHiltunen

File: .travis.invenio.cfg [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+# EXTENSIONS = [
+# 'invenio.ext.sqlalchemy',
+# 'invenio.ext.restful',
+# 'invenio.ext.template',
+# 'flask_breadcrumbs:Breadcrumbs',
+# ]
+
+PACKAGES = [
+    'invenio_base',
```
Comment (jirikuncar) [inline .travis.invenio.cfg]: `invenio_base` should be at the end

**Pull request merged**